cmd/snap/cmd_snap_op.go

```diff
@@ -727,7 +727,7 @@ func (x *cmdInstall) installMany(names []string, opts *client.SnapOptions) error
 		return err
 	}
 
-	if changedSnaps.hasChanges() {
+	if changedSnaps != nil && changedSnaps.hasChanges() {
 		if err := showDone(x.client, chg, changedSnaps, "install", opts, x.getEscapes()); err != nil {
 			return err
 		}
```

Comment:
I would prefer to have the `if changedSnaps == nil` check happen immediately after the check `if err != nil`. If `changedSnaps` is `nil`, then the only thing that can happen is `fmt.Fprintf(Stdout, i18n.G("%s already installed\n"), name)` for each `name` in `names`, but the return will always be `nil`, and I don't think the "already installed" message is correct here. So I suggest the above:

Reply:
I actually would like to print the "<snap> already installed" string, some of the failing tests were checking output strings and otherwise nothing would be shown, which would be strange for users.